### PR TITLE
Add support for DECIMAL input to greatest and least Spark functions

### PR DESCRIPTION
**Pull request by zhli1142015**
Add support for DECIMAL input to greatest and least Spark functions.
Spark adds cast to handle different decimal types during planning, so we don't 
need to handle the type difference in functions.
https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala#L482
Example:
```
spark.sql("""
SELECT *
    FROM VALUES (
        CAST(5.345 AS DECIMAL(6, 2)),
        CAST(5.35 AS DECIMAL(5, 4)))) AS data(a, b);
""").write.mode("overwrite").parquet("decimal_table")
spark.read.parquet("decimal_table").createOrReplaceTempView("decimal_table")
val df = spark.sql("SELECT greatest(a, b) from decimal_table")
```
`Project [greatest(cast(a#6 as decimal(8,4)), cast(b#7 as decimal(8,4))) AS greatest(a, b)#10]`

### CODE REVIEW

**Comment by zhli1142015** [comment]
cc @rui-mo , thanks.

**Review by rui-mo — COMMENTED** [review]
Thanks! Looks good to me. @PHILO-HE Do you have any comment?

File: velox/functions/sparksql/LeastGreatest.cpp [hunk 1]
```diff
@@ -155,6 +156,13 @@ std::vector<std::shared_ptr<exec::FunctionSignature>> leastSignatures() {
                                 .variableArity()
                                 .build());
   }
+  signatures.emplace_back(exec::FunctionSignatureBuilder()
+                              .integerVariable("p")
+                              .integerVariable("s")
+                              .returnType("decimal(p,s)")
+                              .argumentType("decimal(p,s)")
+                              .variableArity()
```
Comment (PHILO-HE) [inline velox/functions/sparksql/LeastGreatest.cpp:164]: As this function requires at least two arguments, I think an extra `.argumentType("decimal(p,s)")` should be added.

Quote from scalar-functions.rst:
_"FunctionSignatureBuilder allows specifying that the last augment may appear zero or more times by calling variableArity() method"_

Comment (zhli1142015) [reply velox/functions/sparksql/LeastGreatest.cpp:164]: Then we need to repeat argumentType for other types also.

Comment (zhli1142015) [reply velox/functions/sparksql/LeastGreatest.cpp:165]: Good catch, Updated, thanks.

**Review by PHILO-HE — APPROVED** [review]
Looks good! Thanks!

**Review by rui-mo — APPROVED** [review]
Thanks.

**Comment by rui-mo** [comment]
@zhli1142015 We need the help from a Meta-employee for further review or action. Thanks.

**Comment by zhli1142015** [comment]
I see, thanks.

**Comment by zhli1142015** [comment]
This PR is approved by @rui-mo and @PHILO-HE . @mbasmanova , @Yuhta and @pedroerp  could you please help review?

**Pull request closed without merging**